content/post/scheduling-in-kubernetes.md

```diff
@@ -0,0 +1,219 @@
+---
+title: "Scheduling in Kubernetes"
+date: 2018-02-01T22:08:22Z
+draft: false
+---
+
+The importance of understanding the implementation of the tools we use in
+production every day cannot be underestimated.
+
+This process informs about the tradeoffs engineers made in the implementations.
+Knowing a tool's strengths and weaknesses helps better design systems on top
+of it; it exposes potential failure modes and helps debug critical errors when
+they occur. It also reveals brilliant ideas, tricks, patterns and conventions used in
+production systems.
+
+In this post, I will go through the implementation of the default scheduler in
+Kubernetes (k8s).
+Although, k8s has a large set of features, tending towards a full-blown PaaS,
+at it’s core, k8s is a cluster scheduler, that is to say, it schedules work
+(pods) onto computing resources (nodes).
+
+## The genericScheduler
+
+K8s has different mechanisms to control pod scheduling: node and pod affinity and
+anti-affinity taints and tolerations and it also has support for custom schedulers. I will look at how all these options factor into the default
```

Comment:
tolerations, (comma) or use dashes instead of : to separate mechanisms from support.